src/classes/BZ_CloseRecruitmentController.apxc

```
@@ -201,151 +131,70 @@ public class BZ_CloseRecruitmentController {
+            m_campaignSnapshot.Volunteer_Opportunities__c=m_campaign.Volunteer_Opportunities__c;
+            m_campaignSnapshot.Sourcing_Info_Options__c=m_campaign.Sourcing_Info_Options__c;
+            //End of the Program Information Fields
+            //Email Fields section
+            m_campaignSnapshot.Invite_Email_Template__c=m_campaign.Invite_Email_Template__c;
+            m_campaignSnapshot.Intro_Email_Template__c=m_campaign.Intro_Email_Template__c;
+            m_campaignSnapshot.App_Open_Email_Template__c=m_campaign.App_Open_Email_Template__c;
+            m_campaignSnapshot.App_Rejected_Email_Template__c=m_campaign.App_Rejected_Email_Template__c;
+            m_campaignSnapshot.App_Submit_Reminder_Template__c=m_campaign.App_Submit_Reminder_Template__c;
+            m_campaignSnapshot.App_Waitlisted_Email_Template__c=m_campaign.App_Waitlisted_Email_Template__c;
+            m_campaignSnapshot.App_Not_Accepted_Email_Template__c=m_campaign.App_Not_Accepted_Email_Template__c;
+            m_campaignSnapshot.App_Interview_Requested_Email_Template__c=m_campaign.App_Interview_Requested_Email_Template__c;
+            m_campaignSnapshot.App_Accepted_Email_Template__c=m_campaign.App_Accepted_Email_Template__c;
+            m_campaignSnapshot.Previous_Candidate_New_Invite__c = prevCandidateEmailTemplate;
+            m_campaignSnapshot.Recruitment_Closed__c=True;
+            m_campaignSnapshot.ProcessType__c='Create';
+            //End of the Email Fields
+            // Check if Campaign is null or not  
+            if(m_campaignSnapshot!=null & m_campaign!=null)
+            {	//Inserting new campaign with the old campaign details
+                insert m_campaignSnapshot;
+                //Updating the old campaign details
+            	update m_campaign;
+            }
+            }
             }
-            System.Debug('BZ_CloseRecruitmentDatabaseChangesQueueable.execute(): end');
+            catch(Exception ex)
```

Why catch and throw? Why not just let it bubble up?

---

What do you mean by bubble up?